corehq/apps/translations/app_translations/upload_app.py

```diff
@@ -248,42 +248,58 @@ class BulkAppTranslationModulesAndFormsUpdater(BulkAppTranslationUpdater):
     def __init__(self, app, lang=None):
         super(BulkAppTranslationModulesAndFormsUpdater, self).__init__(app, lang)
 
+    def get_document_by_legacy_identifier(self, identifier):
+        identifying_parts = identifier.split('_')
+        if len(identifying_parts) not in (1, 2):
+            raise ValueError
+        module_index = int(identifying_parts[0].replace("menu", "").replace("module", "")) - 1
+        document = self.app.get_module(module_index)
+        if len(identifying_parts) == 2:
+            form_index = int(identifying_parts[1].replace("form", "")) - 1
+            document = document.get_form(form_index)
+        return document
+
+    def get_document_by_identifier(self, identifier):
+        unique_id = identifier.split(':')[1]
+        if is_module_sheet(identifier):
+            return self.app.get_module_by_unique_id(unique_id)
+        if is_form_sheet(identifier):
+            return self.app.get_form(unique_id)
+
+    def get_document(self, identifier):
```

Comment:
Naming nitpick: what would you think of `get_module_or_form`? It's clunky, but `get_document` is vague and not really accurate.